phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesStatsTableProvider.java

```diff
@@ -0,0 +1,60 @@
+package org.apache.phoenix.query;
+
+import static org.apache.phoenix.monitoring.GlobalClientMetrics.GLOBAL_HCONNECTIONS_COUNTER;
+
+import java.io.IOException;
+import java.sql.SQLException;
+
+import com.google.common.base.Optional;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.client.Connection;
+import org.apache.hadoop.hbase.client.Table;
+import org.apache.phoenix.exception.SQLExceptionCode;
+import org.apache.phoenix.exception.SQLExceptionInfo;
+import org.apache.phoenix.jdbc.PhoenixDatabaseMetaData;
+import org.apache.phoenix.util.LogUtil;
+import org.apache.phoenix.util.ReadOnlyProps;
+import org.apache.phoenix.util.SchemaUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ConnectionQueryServicesStatsTableProvider implements StatsTableProvider {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(
+            ConnectionQueryServicesStatsTableProvider.class);
+    private final ReadOnlyProps readOnlyProps;
+    private final Configuration config;
+    private final ConnectionQueryServices queryServices;
+
+    ConnectionQueryServicesStatsTableProvider(ConnectionQueryServices queryServices, ReadOnlyProps readOnlyProps, Configuration config){
+        this.queryServices = queryServices;
+        this.readOnlyProps = readOnlyProps;
+        this.config = config;
+    }
+
+    @Override
```

Comment:
You are implementing an interface instead of extending a base class. It seems very strange to me to use @Override here.